substrate/client/network/src/litep2p/peerstore.rs

```diff
@@ -270,21 +274,25 @@ pub struct Peerstore {
 
 impl Peerstore {
 	/// Create new [`Peerstore`].
-	pub fn new(bootnodes: Vec<PeerId>) -> Self {
-		let peerstore_handle = PeerstoreHandle(Arc::new(Mutex::new(Default::default())));
-
-		for bootnode in bootnodes {
-			peerstore_handle.add_known_peer(bootnode);
-		}
-
-		Self { peerstore_handle }
-	}
-
-	/// Create new [`Peerstore`] from a [`PeerstoreHandle`].
-	pub fn from_handle(peerstore_handle: PeerstoreHandle, bootnodes: Vec<PeerId>) -> Self {
-		for bootnode in bootnodes {
-			peerstore_handle.add_known_peer(bootnode);
-		}
+	pub fn new(bootnodes: Vec<PeerId>, metrics_registry: Option<Registry>) -> Self {
+		let metrics = if let Some(registry) = &metrics_registry {
+			PeerSetMetrics::register(registry)
+				.map_err(|err| {
+					log::error!(target: LOG_TARGET, "Failed to register peer set metrics: {}", err);
+					err
+				})
+				.ok()
+		} else {
+			None
+		};
+
+		let peerstore_inner = PeerstoreHandleInner {
```

Comment:
nit: May be add/use a constructor for `PeerstoreHandle` and not initialize `PeerstoreHandleInner` manually?

Reply:
Yep that makes sense, thanks!